db/db_options_test.cc

```diff
@@ -1033,6 +1034,137 @@ TEST_F(DBOptionsTest, SetFIFOCompactionOptions) {
   ASSERT_EQ(fifo_temp_opt[1].age, 30000);
 }
 
+TEST_F(DBOptionsTest, OffPeakTimes) {
+  Options options;
+  options.create_if_missing = true;
+
+  auto verify_invalid = [&]() {
+    Status s = DBImpl::TEST_ValidateOptions(options);
+    ASSERT_NOK(s);
+    ASSERT_TRUE(s.IsInvalidArgument());
+  };
+
+  auto verify_valid = [&]() {
+    Status s = DBImpl::TEST_ValidateOptions(options);
+    ASSERT_OK(s);
+    ASSERT_FALSE(s.IsInvalidArgument());
+  };
+  std::vector<std::string> invalid_cases = {
+      "06:30-",         "-23:30",           // Both need to be set
+      "12:30 PM-23:30", "12:01AM-11:00PM",  // Invalid format
+      "01:99-25:00",  // Invalid value for the hours or minutes
+      "random-value",   "No:No-Hi:Hi",
+  };
+
+  std::vector<std::string> valid_cases = {
+      "",  // Not enabled. Valid case
+      "06:30-11:30",
+      "06:30-23:30",
```

Comment:
Ah, my first version of this PR made it invalid when the start time and end time were separate values, but the validation got removed as I made the change. 

`IsNowOffPeak()` is always going to return `false` in that case and nothing bad really will happen, but still, I think it's safe to add to avoid user confusion.

Reply:
Actually, after another thought, I will leave the same values as valid and it will work as the entire 24 hours are considered offpeak.

Currently, there's no way for users to set the entire 24 hours to be offpeak. There's always going to be 1 minute gap.

Reply:
Yeah, that was a bug and had to be fixed, but I ended up not allowing start_time == end_time and introduced a new boolean option for "always offpeak" Thanks again for the review.